medusa/tv.py

```diff
@@ -1265,71 +1264,55 @@ def load_from_indexer(self, tvapi=None):
         self._save_externals_to_db()
 
     def load_imdb_info(self):
-        """Load all required show information from IMDb with IMDbPY."""
-        imdb_api = imdb.IMDb()
-
-        try:
-            if not self.imdb_id:
-                # Somewhere title2imdbID started to return without 'tt'
-                self.imdb_id = imdb_api.title2imdbID(self.name, kind='tv series')
-
-            if not self.imdb_id:
-                logger.log(u'{0}: Not loading show info from IMDb, '
-                           u"because we don't know its ID".format(self.indexerid))
-                return
-
-            # Make sure we only use one ID, and sanitize the imdb to include the tt.
-            self.imdb_id = self.imdb_id.split(',')[0]
-            if 'tt' not in self.imdb_id:
-                self.imdb_id = 'tt{imdb_id}'.format(imdb_id=self.imdb_id)
-
-            logger.log(u'{0}: Loading show info from IMDb with ID: {1}'.format(
-                self.indexerid, self.imdb_id), logger.DEBUG)
-
-            # Remove first two chars from ID
-            imdb_obj = imdb_api.get_movie(self.imdb_id[2:])
+        """Load all required show information from IMDb with ImdbPie."""
+        imdbapi = imdbpie.Imdb()
+
+        if not self.imdb_id:
+            imdb_results = imdbapi.search_for_title(self.name)
+            for result in imdb_results:
+                imdb_id = result['imdb_id']
+                title = result['title']
+                year = result['year']
+                logger.log(u'Results: {0} {1} {2}'.format( title, year, imdb_id))
+                if title == self.name and year == self.startyear:
+                    logger.log(u'{0} {1} matches IMDb year {2} and title {3}, using {4} as the imdb_id for the show.'.format(self.name, self.startyear, title, year, imdb_id))
+                    self.imdb_id = imdb_id
+                elif title == self.name and year != self.startyear and not self.imdb_id:
+                    logger.log(u'{0} matches IMDb title {1} , using {2} as the imdb_id for the show.'.format(self.name, title, imdb_id))
+                    self.imdb_id = imdb_id
```

Comment:
This won't always work. Take as example the search for "X Factor". It will return:
```
{u'year': u'2011', u'imdb_id': u'tt1582461', u'title': u'The X Factor'}
{u'year': u'2008', u'imdb_id': u'tt1188980', u'title': u'X Factor'}
{u'year': u'2010', u'imdb_id': u'tt1709082', u'title': u'X Factor'}
{u'year': u'2006', u'imdb_id': u'tt0892254', u'title': u'X Factor'}
{u'year': u'2008', u'imdb_id': u'tt1194223', u'title': u'X Factor'}
{u'year': u'2011', u'imdb_id': u'tt1774260', u'title': u'X Factor'}
```
I want the X Factor from year 2006, but the loop will keep changing it until it gets to X Factor from year 2011. Also the title comparison should be case-insensitive and it's probably a good idea to make it a method like it was before e.g. ```title_to_imdb(titles)```.